core/unit_test/TestGraph.hpp

```diff
@@ -564,9 +564,20 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), end_of_submit_control_flow) {
   //      their policy if the defaulted graph implementation is used.
   graph.submit(exec_3);
 
+// GCC 10.1 is buggy for CTAD. Interestingly, it fails only for the CTAD that is
+// needed "in a" Kokkos::parallel_for. For those needed "in a" then_parallel_for,
+// it's fine.
+// https://stackoverflow.com/questions/63493436/ctad-and-template-constructor-with-member-with-parameter-pack
+// https://github.com/kokkos/kokkos/issues/7316
+#if defined(KOKKOS_COMPILER_GNU) && (1010 == KOKKOS_COMPILER_GNU)
+  #define GCC_1010_CTAD(data, index, num) <decltype(data), index, num>
+#else
+  #define GCC_1010_CTAD(...)
+#endif
+
   Kokkos::parallel_for(
       policy_t(exec_3, 0, 1),
-      FetchValuesAndContribute(data, {index_D(), index_E()}, index_F, value_F));
+      FetchValuesAndContribute GCC_1010_CTAD(data, index_F, 2)(data, {index_D(), index_E()}, index_F, value_F));
```

Comment:
Considering that this is only used in one place I would prefer
```suggestion
#if defined(KOKKOS_COMPILER_GNU) && (1010 == KOKKOS_COMPILER_GNU)
      // Workaround CTAD bug
      FetchValuesAndContribute<data, index_F, 2>(data, {index_D(), index_E()}, index_F, value_F));
#else
      FetchValuesAndContribute(data, {index_D(), index_E()}, index_F, value_F));
#endif
```

Reply:
:checkered_flag: